sqlglot/generator.py

```diff
@@ -499,8 +503,8 @@ def identifier_sql(self, expression):
     def partition_sql(self, expression):
         keys = csv(
             *[
-                f"{k.args['this']}='{v.args['this']}'" if v else k.args["this"]
-                for k, v in expression.args.get("this")
+                f"{prop.this.this}={prop.args['value']}" if prop.args["value"] else prop.this.this
```

Comment:
is this right?

why not 

prop.text("this") or prop.name

and prop.text("value")